File: DesignPatternsInAutomatedTesting-Series/PatternsInAutomation.Tests/Advanced/Decorator/Advanced/Strategies/TotalPriceOrderPurchaseStrategy.cs
using PatternsInAutomatedTests.Advanced.Decorator.Pages.PlaceOrderPage;

namespace PatternsInAutomatedTests.Advanced.Decorator.Advanced.Strategies
{
    public class TotalPriceOrderPurchaseStrategy : OrderPurchaseStrategy
    {
        private readonly decimal itemsPrice;

        public TotalPriceOrderPurchaseStrategy(decimal itemsPrice)
        {
            this.itemsPrice = itemsPrice;
        }

        public override decimal CalculateTotalPrice()
        {
            return itemsPrice;
        }

        public override void ValidateOrderSummary(decimal totalPrice)
        {
            PlaceOrderPage.Instance.Validate().OrderTotalPrice(totalPrice.ToString());
        }
    }
}

File: DesignPatternsInAutomatedTestingJava-Series/src/test/java/decorator/advanced/strategies/TotalPriceOrderPurchaseStrategy.java
package decorator.advanced.strategies;

import decorator.core.Driver;
import decorator.pages.checkoutpage.CheckoutPage;

import java.math.BigDecimal;

public class TotalPriceOrderPurchaseStrategy extends OrderPurchaseStrategy {
    private final BigDecimal itemsPrice;

    public TotalPriceOrderPurchaseStrategy(BigDecimal itemsPrice) {
        this.itemsPrice = itemsPrice;
    }

    @Override
    public BigDecimal calculateTotalPrice() {
        return itemsPrice;
    }

    @Override
    public void validateOrderSummary(BigDecimal totalPrice) {
        Driver.retry(10,500, ()->Driver.getPage(CheckoutPage.class).assertions()
                .assertOrderTotalPrice(totalPrice));
    }
}
